### ISSUE TITLE
Default value taken from "Model" for JSONSchema oneOf

### ISSUE THREAD

**Issue by navnita-170310093** [ngx-formly/ngx-formly#3395]
**Description**
Within the JSONSchema oneOf, the default value taken from model is used only at the initialization. When the oneOf choice is changed the default value is not set.

**Minimal Reproduction**
Here is a modified example of the official JSONSchema example:

https://angular-1dt17d.stackblitz.io/

Change the oneOf choice, the default value is not in the input field.

The schema looks like this
`{
  "schema": {
    "type": "object",
    "properties": {
      "Form": {
        "title": "Form",
        "oneOf": [
          {
            "title": "Option 1",
            "properties": {
              "Name": {
                "title": "Name",
                "type": "string"
              }
            },
            "type": "object"
          },
          {
            "title": "Option 2",
            "items": {
              "properties": {
                "Name": {
                  "title": "Name",
                  "type": "string"
                }
              },
              "type": "object"
            },
            "type": "array"
          }
        ]
      }
    }
  },
  "model": {
    "Form": [{ "Name": "Navnita" }, { "Name": "Tanu" }, { "Name": "Swii" }]
  }
}
`

**Comment by aitboudad** [comment]
Its an expected behavior, the `model` doesn't represent the default value of the field. use `default` property instead.

**Comment by navnita-170310093** [comment]
I need one solution for this.
Our requirement is like this...

Default value is in the form of JSON. So, I added the default json in model. 
I can't manually assign default property in schema because schema is huge and also the jsonSchema is generating from xsd. Nothing we are changing manually

Angular formly document says:
    -   There are two ways to initialize the form with default values.
    -   Default the model (preferred)
    -   Use the defaultValue option on the field

Now, I want json default value should be dispayed in the formly UI. How to do this because when we change oneOf choice, The default value is not set.

If this issue is not with defaultValue option, then we want one solution also for **model**

**Comment by aitboudad** [comment]
> There are two ways to initialize the form with default values.

Yes that's true, it says initialize value but it doesn't means that passing the value through the `model` has the same behavior as `default` property.

> How to do this because when we change oneOf choice, The default value is not set.

The preferred way is `default` property, another solution is subscribe the the model changes and set the default value manually.
